proj-android/PowerAuthLibrary/src/main/java/io/getlime/security/powerauth/networking/client/HttpClientTask.java

```diff
@@ -0,0 +1,197 @@
+    private byte[] loadBytesFromInputStream(InputStream is) throws IOException {
+        if (is == null) {
+            return null;
+        }
+        ByteArrayOutputStream result = new ByteArrayOutputStream();
+        byte[] buffer = new byte[1024];
+        int length;
+        while ((length = is.read(buffer)) != -1) {
+            result.write(buffer, 0, length);
+            if (isCancelled()) {
+                return null;
+            }
+        }
+        return result.toByteArray();
+    }
+
+    @Override
+    protected TResponse doInBackground(TRequest... tRequests) {
+        try {
+            if (isCancelled()) {
+                return null;
+            }
+
+            // Prepare request data
+            HttpRequestHelper.RequestData requestData = httpRequestHelper.buildRequest(baseUrl, cryptoHelper);
+
+            // Create an URL connection
+            final HttpURLConnection urlConnection = (HttpURLConnection) requestData.url.openConnection();
+            final boolean securedUrlConnection = urlConnection instanceof HttpsURLConnection;
+
+            // Setup the connection
+            urlConnection.setRequestMethod(requestData.method);
+            urlConnection.setDoOutput(true);
+            urlConnection.setUseCaches(false);
+            urlConnection.setConnectTimeout(clientConfiguration.getConnectionTimeout());
+            urlConnection.setReadTimeout(clientConfiguration.getReadTimeout());
+            for (Map.Entry<String, String> header : requestData.httpHeaders.entrySet()) {
+                urlConnection.setRequestProperty(header.getKey(), header.getValue());
+            }
+
+            // ssl validation strategy
+            if (securedUrlConnection) {
+                final PA2ClientValidationStrategy clientValidationStrategy = clientConfiguration.getClientValidationStrategy();
+                if (clientValidationStrategy != null) {
+                    final HttpsURLConnection sslConnection = (HttpsURLConnection) urlConnection;
+                    final SSLSocketFactory sslSocketFactory = clientValidationStrategy.getSSLSocketFactory();
+                    if (sslSocketFactory != null) {
+                        sslConnection.setSSLSocketFactory(sslSocketFactory);
+                    }
+                    final HostnameVerifier hostnameVerifier = clientValidationStrategy.getHostnameVerifier();
+                    if (hostnameVerifier != null) {
+                        sslConnection.setHostnameVerifier(hostnameVerifier);
+                    }
+                }
+            } else {
+                if (!clientConfiguration.isUnsecuredConnectionAllowed()) {
+                    throw new SSLException("Connection to non-TLS endpoint is not allowed.");
+                }
+            }
+
+            // Connect to endpoint
+            if (requestData.body != null) {
+                urlConnection.getOutputStream().write(requestData.body);
+            }
+            urlConnection.connect();
```

Comment:
Connection is not disconnected.
https://developer.android.com/reference/java/net/HttpURLConnection